addons/udes_advanced_packaging/tests/test_package_reservation.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+from odoo.exceptions import ValidationError
+from odoo.addons.udes_stock.tests.common import BaseUDES
+
+
+class TestPackageReservation(BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestPackageReservation, cls).setUpClass()
+        # enable full package reservation
+        cls.picking_type_pick.u_reserve_as_packages = True
+
+    def test_reserve_full_package_one_product(self):
+        """Test to reserve a full package for one product
+        when initial demand < package quantity
+        """
+
+        # create a package
+        package = self.create_package()
+
+        # create a quant of 4 apples in a stock sublocation for the package
+        quant = self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, 4, package_id=package.id
+        )
+        # the quant should be unreserved
+        self.assertEqual(quant.reserved_quantity, 0, "The quant is reserved")
```

Comment:
Nit can we drop the doc string - the error message from the asser Equal is probably more useful

Reply:
"Test to see if the quant is reserved?"

Reply:
I would just remove it, self.assertEqual(quant.reserved_quantity, 0)
it will spit out 5!=0 which is clear, 
otherwise the doc string needs to do something similar
"Expected 0 reserved quantity, got {quant.reserved_quantity}"

Reply:
@caleb-shelton think this one still needs doing